docs/manual/working/scalaGuide/main/sql/slick/code/GlobalLookup.scala

```diff
@@ -6,22 +6,23 @@ package global
 
 import scala.concurrent.Future
 
-import play.api.Play
+import javax.inject.{Inject, Provider}
+import play.api.Application
 import play.api.mvc._
 import play.api.db.slick.DatabaseConfigProvider
 import play.api.libs.concurrent.Execution.Implicits.defaultContext
 
-import slick.jdbc.JdbcProfile
+import slick.driver.JdbcProfile
 
 import UsersSchema._
 
-object Application extends Controller {
+class Application1 @Inject()(provider: Provider[Application]) extends Controller {
```

Comment:
You can directly inject the `Application` itself. This will be more direct and clear.